solarkraft/src/fetcher/storage.ts

```diff
@@ -62,3 +67,68 @@ export interface ContractCallEntry {
      */
     oldFields: FieldsMap
 }
+
+/**
+ * Store a contract call entry in the file storage.
+ * @param root the storage root directory
+ * @param entry a call entry
+ */
+export function saveContractCallEntry(root: string, entry: ContractCallEntry) {
+    const filename = getEntryFilename(root, entry)
+    // convert OrderedMaps to arrays
+    const simplified = {
+        ...entry,
+        fields: entry.fields.toArray(),
+        oldFields: entry.oldFields.toArray(),
+    }
+    const contents = JSONbig.stringify(simplified)
+    writeFileSync(filename, contents)
+    return filename
+}
+
+/**
+ * Load a contract call entry in the file storage.
+ * @param root the storage root directory
+ * @param entry a call entry
+ * @returns the loaded call entry
+ */
+export function loadContractCallEntry(filename: string): ContractCallEntry {
+    const contents = readFileSync(filename)
+    const loaded = JSONbig.parse(contents)
+    return {
+        height: loaded.height as number,
+        contractId: loaded.contractId as string,
+        txHash: loaded.txHash as string,
+        method: loaded.method as string,
+        methodArgs: loaded.methodArgs as any[],
+        returnValue: loaded.returnValue,
+        fields: OrderedMap<string, any>(loaded.fields),
+        oldFields: OrderedMap<string, any>(loaded.oldFields),
+    }
```

Comment:
I think I'd prefer to have a validation method, which checks that `loaded` has the correct shape, and then returns the value cast to the correct type with given field types, instead of just blind-casting each field.

Reply:
Please feel free to implement one!